app/views/shared/_table.html.erb

```diff
@@ -24,7 +25,21 @@
  window.onload = function() {
    $('#data-table').DataTable({
      "order": [[ 1, "desc" ]],
-     "stateSave": true
+     "stateSave": true,
+     "stateSaveParams": function (settings, data) {
+       const savableKeys = ["search"]
+       Object.keys(data).forEach(function(key) {
+         if (!savableKeys.includes(key)) {
+           delete data[key]
```

Comment:
Note, I had to do it this way where we `delete` out of the original `data` object instead of just creating a new one as to not break the object reference in the callback.